lib/Migrator/APIDiffMigratorPass.cpp

```diff
@@ -1290,6 +1292,94 @@ struct APIDiffMigratorPass : public ASTMigratorPass, public SourceEntityWalker {
     }
   }
 
+  llvm::StringSet<> funcNamesForOverrideRemoval() {
+    llvm::StringSet<> Results;
+    Results.insert("application(_,delegateHandlesKey)");
+    Results.insert("changeColor(_)");
+    Results.insert("controlTextDidBeginEditing(_)");
+    Results.insert("controlTextDidEndEditing(_)");
+    Results.insert("controlTextDidChange(_)");
+    Results.insert("changeFont(_)");
+    Results.insert("validModesForFontPanel(_)");
+    Results.insert("discardEditing()");
+    Results.insert("commitEditing()");
+    Results.insert("commitEditing(withDelegate,didCommit,contextInfo)");
+    Results.insert("commitEditingAndReturnError()");
+    Results.insert("objectDidBeginEditing(_)");
+    Results.insert("objectDidEndEditing(_)");
+    Results.insert("validateMenuItem(_)");
+    Results.insert("pasteboard(_,provideDataForType)");
+    Results.insert("pasteboardChangedOwner(_)");
+    Results.insert("validateToolbarItem(_)");
+    Results.insert("layer(_,shouldInheritContentsScale,from)");
+    Results.insert("view(_,stringForToolTip,point,userData)");
+    return Results;
+  }
+
+  SourceLoc shouldRemoveOverride(AbstractFunctionDecl *AFD) {
```

Comment:
Is this based purely on the function name at the moment?

If it is, it seems like this migration would cause errors if
1) I have my own class with a similarly named, but unrelated method, e.g. 'changeColor(_ color: Color)',  that I override in subclass (the subclass' override should be left alone)
2) I have a subclass of NSObject and a subclass of that subclass that both override one of these methods (the sub-sub-class' method should still override).

Some of the names seem specific enough that the above wouldn't happen too often in practice, but I could imagine it happening for a few of them.

Reply:
yeah, I think it is safer to use USR for identifying these overriding function. Will change!